src/internal_modules/roc_sndio/wav_sink.cpp

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright (c) 2023 Roc Streaming authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include "roc_sndio/wav_sink.h"
+#include "roc_core/endian_ops.h"
+#include "roc_core/log.h"
+#include "roc_core/panic.h"
+#include "roc_sndio/backend_map.h"
+
+namespace roc {
+namespace sndio {
+
+WavSink::WavSink(core::IArena& arena, const Config& config)
+    : output_file_(NULL)
+    , header_(config.sample_spec.num_channels(), config.sample_spec.sample_rate(), 32)
+    , buffer_(arena)
+    , buffer_size_(0)
+    , valid_(false) {
+    BackendMap::instance();
```

Comment:
It's not needed but it seemed to be not related specifically to sox, so it's been also invoked here. It could've been noticed that it's not used for PulseAudio and removed if everything works just fine without it.